### PR TITLE
docs: add netlify-cms-backend-* readme for developers

### PR DESCRIPTION
**Pull request by NikitaIT**
**Summary**

Added short readme. It's better than nothing. Now these packages are of no value in and of themselves, so I think the "quick look at the code" format is appropriate.

**Test plan**

I dont know)

**A picture of a cute animal**
sad dog

### CODE REVIEW

File: packages/netlify-cms-backend-git-gateway/README.md [hunk 1]
```diff
@@ -1,11 +1,20 @@
-# Docs coming soon!
+# Git gateway
```
Comment (erezrokah) [inline packages/netlify-cms-backend-git-gateway/README.md:1]:
```suggestion
# Git Gateway
```

Per https://docs.netlify.com/visitor-access/git-gateway/

File: packages/netlify-cms-backend-git-gateway/README.md [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
 
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
+This is `all git backends with Netlify` file manager.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-git-gateway/README.md:3]:
```suggestion
Netlify's gateway to hosted git APIs
```

Per https://github.com/netlify/git-gateway

File: packages/netlify-cms-backend-azure/README.md [hunk 3]
```diff
@@ -1,11 +1,18 @@
-# Docs coming soon!
+# Backend Azure
 
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
+This is `azure REST-API` file manager.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:3]:
```suggestion
An abstraction layer between the CMS and Azure DevOps
```

File: packages/netlify-cms-backend-azure/README.md [hunk 4]
```diff
@@ -1,11 +1,18 @@
-# Docs coming soon!
+# Backend Azure
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:1]:
```suggestion
# Azure backend
```

File: packages/netlify-cms-backend-azure/README.md [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
 
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
+This is `azure REST-API` file manager.
 
-In the meantime, you can:
+## Has `Api`, `Backend` and `AuthenticationPage`.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:5]:
```suggestion
## Code structure
```

File: packages/netlify-cms-backend-azure/README.md [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
 
-1. Check out the [main readme](https://github.com/netlify/netlify-cms/#readme) or the [documentation
-   site](https://www.netlifycms.org) for more info.
-2. Reach out to the [community chat](https://netlifycms.org/chat/) if you need help.
-3. Help out and [write the readme yourself](https://github.com/netlify/netlify-cms/edit/master/packages/netlify-cms-backend-azure/README.md)!
+`Api` - wrapper for azure `REST-API`.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:7]:
```suggestion
`Api` - A wrapper for Azure REST API.
```

File: packages/netlify-cms-backend-azure/README.md (hunk 3 above)
Comment (erezrokah) [reply packages/netlify-cms-backend-azure/README.md:3]: "file manager" can be confusing as the backends handle additional `git` related operations

File: packages/netlify-cms-backend-azure/README.md [hunk 7]
```diff
@@ -1,11 +1,18 @@
...
+
+`AuthenticationPage` is component for [lib-auth](https://github.com/netlify/netlify-cms/edit/master/packages/netlify-cms-lib-auth/README.md) login by:
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:9]: I'm not sure I understand this sentence.
Maybe:
> Facilitates Azure authentication flow

File: packages/netlify-cms-backend-proxy/README.md [hunk 8]
```diff
@@ -1,11 +1,13 @@
-# Docs coming soon!
+# Backend Proxy
 
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
+This is `RPC-proxy` file manager.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-proxy/README.md:3]:
```suggestion
Facilitates [local development ](https://www.netlifycms.org/docs/beta-features/#working-with-a-local-git-repository)
```

File: packages/netlify-cms-backend-test/README.md [hunk 9]
```diff
@@ -1,11 +1,20 @@
-# Docs coming soon!
+# Backend Test
 
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
+This is `in-memory` file manager.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-test/README.md:3]: The backend behind https://cms-demo.netlify.com/.
Used for demo purposes only

File: packages/netlify-cms-lib-auth/README.md [hunk 10]
```diff
@@ -1,11 +1,3 @@
-# Docs coming soon!
+# Backend Lib Auth
```
Comment (erezrokah) [inline packages/netlify-cms-lib-auth/README.md:1]:
```suggestion
# Lib Auth
```

File: packages/netlify-cms-lib-auth/README.md [hunk 11]
```diff
@@ -1,11 +1,3 @@
...
-Netlify CMS was recently converted from a single npm package to a "monorepo" of over 20 packages.
-That's over 20 Readme's! We haven't created one for this package yet, but we will soon.
-
-In the meantime, you can:
-
-1. Check out the [main readme](https://github.com/netlify/netlify-cms/#readme) or the [documentation
-   site](https://www.netlifycms.org) for more info.
-2. Reach out to the [community chat](https://netlifycms.org/chat/) if you need help.
-3. Help out and [write the readme yourself](https://github.com/netlify/netlify-cms/edit/master/packages/netlify-cms-lib-auth/README.md)!
+This is auth router for `AuthenticationPage` in `netlify-cms-backend-*`.
```
Comment (erezrokah) [inline packages/netlify-cms-lib-auth/README.md:3]:
```suggestion
Shared components to handle OAuth and implicit authentication flows
```

File: packages/netlify-cms-lib-util/README.md [hunk 12]
```diff
@@ -9,3 +9,7 @@ In the meantime, you can:
    site](https://www.netlifycms.org) for more info.
 2. Reach out to the [community chat](https://netlifycms.org/chat/) if you need help.
 3. Help out and [write the readme yourself](https://github.com/netlify/netlify-cms/edit/master/packages/netlify-cms-lib-util/README.md)!
+
+## Domain-specific file manager
```
Comment (erezrokah) [inline packages/netlify-cms-lib-util/README.md:13]: Should we remove the "# Docs coming soon!" part?

File: packages/netlify-cms-lib-util/README.md [hunk 13]
```diff
@@ -9,3 +9,7 @@ In the meantime, you can:
...
+
+Known as `Implementation` - abstract type for all `Backend` in `netlify-cms-backend-*`. It is a set of high-level methods for managing the Repository and Media files.
```
Comment (erezrokah) [inline packages/netlify-cms-lib-util/README.md:15]:
```suggestion
Shared utilities to handle various `netlify-cms-backend-*` backends operations
```

File: packages/netlify-cms-backend-azure/README.md [hunk 14]
```diff
@@ -1,11 +1,18 @@
...
+- azure app endpoint, from `config.backend`.
+
+`Backend` is domain-specific file manager based on `Api`.
```
Comment (erezrokah) [inline packages/netlify-cms-backend-azure/README.md:12]:
```suggestion
`implementation` - An implementation layer over the Azure DevOps API 
```

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @NickStees, this is very important and we've wanting to update these for a long time.
 I haven't gone through all the changes, but added comments that are relevant for all backends.
 
 Also:
 1. github -> GitHub
 2. bitbucket -> Bitbucket
 3. gitlab -> GitLab
 4. azure -> Azure DevOps
 5. `REST-API` -> REST API (no need to codify)
 6. No need to codify `gitlab`. e.g. `gitlab REST-API` -> GitLab REST API

File: packages/netlify-cms-backend-git-gateway/README.md (hunk 1 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-git-gateway/README.md]: fixed

File: packages/netlify-cms-backend-git-gateway/README.md (hunk 2 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-git-gateway/README.md]: fixed

File: packages/netlify-cms-backend-azure/README.md (hunk 4 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: fixed

File: packages/netlify-cms-backend-azure/README.md (hunk 5 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: fixed

File: packages/netlify-cms-backend-azure/README.md (hunk 6 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: fixed

File: packages/netlify-cms-backend-azure/README.md (hunk 7 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: fixed, but i did not remove the link(lib-auth), imho it is should be here

File: packages/netlify-cms-backend-proxy/README.md (hunk 8 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-proxy/README.md]: fixed

File: packages/netlify-cms-backend-test/README.md (hunk 9 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-test/README.md]: fixed

File: packages/netlify-cms-lib-auth/README.md (hunk 11 above)
Comment (NikitaIT) [reply packages/netlify-cms-lib-auth/README.md]: fixed

File: packages/netlify-cms-lib-auth/README.md (hunk 10 above)
Comment (NikitaIT) [reply packages/netlify-cms-lib-auth/README.md]: fixed

File: packages/netlify-cms-lib-util/README.md (hunk 12 above)
Comment (NikitaIT) [reply packages/netlify-cms-lib-util/README.md]: hm, removed

File: packages/netlify-cms-lib-util/README.md (hunk 13 above)
Comment (NikitaIT) [reply packages/netlify-cms-lib-util/README.md]: fixed

File: packages/netlify-cms-backend-azure/README.md (hunk 14 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: rewriten

File: packages/netlify-cms-backend-azure/README.md (hunk 3 above)
Comment (NikitaIT) [reply packages/netlify-cms-backend-azure/README.md]: Yes you are right. Fixed, but...
I wanted to get away from the current inconsistent naming convention in code and Implementaion is not a name at all.

### My logic is that:
If Git is Source Code Management System and Gitlab/etc is Git-repository managers.
Then `Implementaion` (from `lib-utils`) is File Management System API. Or file manager for short. I changed it to a long name, I don't even know what to call it better.
Вecause:
- This is API, it allows to include authentication methods
- This is Manager (as Composite pattern, in a sense)
- This works with Files (Entry/Media), and git is just a implementation detail

Additional `git` related operations hidden for core client and `Backend` name is relative to the use case. It is only understandable in the `cms-core` context. File Management System API used as Backend for `cms-core`.

**Comment by NikitaIT** [comment]
@erezrokah Thanks for the review. MR improved.

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
🚀

**Pull request merged**